### PR TITLE
larcore 9.11.01 and dependency on buildtools for earlier

### PR DESCRIPTION
**Pull request by marcmengel**
Versions and version related dependency.

### CODE REVIEW

File: packages/larcore/package.py [hunk 1]
```diff
@@ -90,6 +93,7 @@ def fetch_remote_versions(self, concurrency=None):
     depends_on("larcorealg")
     depends_on("art-root-io")
     depends_on("cetmodules", type="build")
+    depends_on("cetbuildtools", type="build" , when="@:09.11")
```
Comment (knoepfel) [inline packages/larcore/package.py]: Looks to me like this should be 

```python
depends_on("cetbuildtools", type="build" , when="@:09.05.00")
```

based on the [commit history](https://github.com/LArSoft/larcore/commit/bac1d631aa01fc4df11370566221ea4f8167967f).

Comment (marcmengel) [reply packages/larcore/package.py]: But it was the failure to build 09.10.01 that prompted the change...

Comment (knoepfel) [reply packages/larcore/package.py]: @marcmengel, that's weird.  Nothing in version `09.10.01` depends on `cetbuildtools`.  Maybe it's an underlying dependency somewhere?

Comment (marcmengel) [reply packages/larcore/package.py]: Okay, I think I see the core of my confusion; the build I was doing that failed was in fact something tagged v09_30_00_rc1 which I suspect was a mistag as it is ahead of the latest release...  But of course spack picked it as the "latest"... So then I put the dependendency in on cetbuildtools, tested *with a specific version* and things "got better" ... So I'm going to put the dependency change where you  mentioned, and take that rc1 tag *out* of the recipe.

Comment (knoepfel) [reply packages/larcore/package.py]: Thanks for sorting that out, @marcmengel.  I think this implies that LArSoft should use a different versioning convention for release candidates.

**Review by knoepfel — APPROVED** [review]

**Pull request merged**